cmd/minishift/cmd/start.go

```diff
@@ -231,18 +235,76 @@ func postClusterUp(machineName string, ip string, port int, routingSuffix string
 		atexit.Exit(1)
 	}
 
-	applyAddOns(ip, routingSuffix, ocPath, kubeConfigPath, sshCommander)
+	addOnManager := addon.GetAddOnManager()
+	configurePersistentVolumes(addOnManager, sshCommander, ocRunner)
+	applyAddOns(addOnManager, ip, routingSuffix, ocPath, kubeConfigPath, sshCommander)
 }
 
-func applyAddOns(ip string, routingSuffix string, ocPath string, kubeConfigPath string, sshCommander provision.SSHCommander) {
-	addOnManager := addon.GetAddOnManager()
+func applyAddOns(addOnManager *manager.AddOnManager, ip string, routingSuffix string, ocPath string, kubeConfigPath string, sshCommander provision.SSHCommander) {
 	err := addOnManager.Apply(addon.GetExecutionContext(ip, routingSuffix, ocPath, kubeConfigPath, sshCommander))
 	if err != nil {
 		fmt.Println("Error executing addon commands: ", err)
 		atexit.Exit(1)
 	}
 }
 
+// TODO - persistent volume creation should really be fixed upstream, aka 'cluster up'. See https://github.com/openshift/origin/issues/14076 (HF)
+// configurePersistentVolumes makes sure that the default persistent volumes created by 'cluster up' have the right permissions - see https://github.com/minishift/minishift/issues/856
+func configurePersistentVolumes(addOnManager *manager.AddOnManager, sshCommander provision.SSHCommander, ocRunner *oc.OcRunner) error {
+	// don't apply this if anyuid is not enabled
+	anyuid := addOnManager.Get("anyuid")
+	if anyuid == nil || !anyuid.IsEnabled() {
+		return nil
+	}
+
+	fmt.Print("-- Waiting for persistent volumes to be created ... ")
+
+	hostPvDir := viper.GetString(hostPvDir)
+
+	var out, err *bytes.Buffer
+
+	// pull the status of the persistent-volume-setup job to determine when the persitent volume creates is completed
+	for {
```

Comment:
@csrwng @LalatenduMohanty @budhrg - here is the updated version of _ configurePersistentVolumes_, which does not rely on knowing the number of persistent volumes to create. Instead it polls the OpenShift job which creates these directories. 

This is obviously a bit neater, the bottom line stays the same.